assets/app/scripts/controllers/overview.js

```diff
@@ -69,6 +73,23 @@ angular.module('openshiftConsole')
       Logger.log("services (list)", $scope.services);
     }));
 
+    watches.push(DataService.watch("routes", $scope, function(routes) {
+      var routeMap = $scope.routesByService;
+      var displayRouteMap = $scope.displayRouteByService;
+      angular.forEach(routes.by("metadata.name"), function(route, routeName){
+        var serviceName = route.serviceName;
+        routeMap[serviceName] = routeMap[serviceName] || {};
+        routeMap[serviceName][routeName] = route;
+
+        // Find the best route to display for a service.
+        if (!displayRouteMap[serviceName] || isCustomHost(route)) {
```

Comment:
This would select the first custom route, not the last:
```
if (!displayRouteMap[serviceName] || (isCustomHost(route) && !isCustomHost(displayRouteMap[serviceName]))) {
```